### PR TITLE
[perf] websocket optimisation

### PR DESCRIPTION
**Pull request by billouboq**
### The kind of change this PR does introduce

* [ ] a bug fix
* [ ] a new feature
* [ ] an update to the documentation
* [x] a code change that improves performance
* [ ] other

### Current behaviour
use forEach loop and anonymous functions

### New behaviour
use for loop and cached function

### Other information (e.g. related issues)
Why it's better to use cached functions:  [see this](https://github.com/uWebSockets/uWebSockets/wiki/The-importance-of-JavaScript-scope-control)

### CODE REVIEW

File: lib/transports/websocket.js [hunk 1]
```diff
@@ -86,31 +88,37 @@ WebSocket.prototype.onData = function (data) {
...
 
-      self.writable = false;
-      self.socket.send(data, opts, function (err) {
-        if (err) return self.onError('write error', err.stack);
-        self.writable = true;
-        self.emit('drain');
-      });
-    });
-  });
+  function sendCb (err) {
```
Comment (darrachequesne) [inline lib/transports/websocket.js]: Same hère, how about something like `onEnd`?

**Comment by darrachequesne** [comment]
Hi, thanks for that PR! I made some minor comments, could you please check them?

Did you see any other area of improvement?

**Comment by billouboq** [comment]
Will check tonight and change what you suggested

**Comment by darrachequesne** [comment]
Thanks!

**Pull request merged**